conans/client/generators/cmake_common.py

```diff
@@ -577,6 +592,36 @@ def generate_targets_section(dependencies):
 endmacro()
 """
 
+apple_frameworks_macro = """
+macro(conan_find_apple_frameworks frameworks)
+    if(APPLE)
+        if(CMAKE_BUILD_TYPE)
+            if(${CMAKE_BUILD_TYPE} MATCHES "Debug")
+                set(CONAN_FRAMEWORKS ${CONAN_FRAMEWORKS_DEBUG} ${CONAN_FRAMEWORKS})
+                set(CONAN_FRAMEWORK_DIRS ${CONAN_FRAMEWORK_DIRS_DEBUG} ${CONAN_FRAMEWORK_DIRS})
+            elseif(${CMAKE_BUILD_TYPE} MATCHES "Release")
+                set(CONAN_FRAMEWORKS ${CONAN_FRAMEWORKS_RELEASE} ${CONAN_BUILD_MODULES_PATHS})
+                set(CONAN_FRAMEWORK_DIRS ${CONAN_FRAMEWORK_DIRS_RELEASE} ${CONAN_FRAMEWORK_DIRS})
+            elseif(${CMAKE_BUILD_TYPE} MATCHES "RelWithDebInfo")
+                set(CONAN_FRAMEWORKS ${CONAN_FRAMEWORKS_RELWITHDEBINFO} ${CONAN_BUILD_MODULES_PATHS})
+                set(CONAN_FRAMEWORK_DIRS ${CONAN_FRAMEWORK_DIRS_RELWITHDEBINFO} ${CONAN_FRAMEWORK_DIRS})
+            elseif(${CMAKE_BUILD_TYPE} MATCHES "MinSizeRel")
+                set(CONAN_FRAMEWORKS ${CONAN_FRAMEWORKS_MINSIZEREL} ${CONAN_BUILD_MODULES_PATHS})
+                set(CONAN_FRAMEWORK_DIRS ${CONAN_FRAMEWORK_DIRS_MINSIZEREL} ${CONAN_FRAMEWORK_DIRS})
+            endif()
+        endif()
+        
+        foreach(_FRAMEWORK ${CONAN_FRAMEWORKS})
+            # https://cmake.org/pipermail/cmake-developers/2017-August/030199.html
+            find_library(CONAN_FRAMEWORK_FOUND NAME ${_FRAMEWORK} PATHS ${CONAN_FRAMEWORK_DIRS})
+            if(CONAN_FRAMEWORK_FOUND)
```

Comment:
What if not found?

Reply:
Added an error message in case the framework is not found